test/xpu_api/tuple/tuple.tuple/tuple.rel/eq.pass.cpp

```diff
@@ -0,0 +1,303 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/tuple>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+class KernelTupleEQTest1;
+class KernelTupleEQTest2;
+
+void
+kernel_test1(sycl::queue& deviceQueue)
+{
+    bool ret = true;
+    sycl::range<1> numOfItems{1};
+    sycl::buffer<bool, 1> buffer1(&ret, numOfItems);
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        auto ret_access = buffer1.get_access<sycl::access::mode::write>(cgh);
+        cgh.single_task<class KernelTupleEQTest1>([=]() {
+            {
+                const dpl::tuple<> t1;
+                const dpl::tuple<> t2;
+
+                ret_access[0] &= (t1 == t2);
+                ret_access[0] &= (!(t1 != t2));
+            }
+            {
+                const dpl::tuple<int> t1(1);
+                const dpl::tuple<float> t2(1.1f);
+
+                ret_access[0] &= (!(t1 == t2));
+                ret_access[0] &= (t1 != t2);
+            }
+            {
+                typedef dpl::tuple<int> T1;
+                typedef dpl::tuple<float> T2;
+                const T1 t1(1);
+                const T2 t2(1.f);

```

Comment:
Please use types of `T1` and `T2` directly everywhere in this file, not only for the first two appearances.

Reply:
Done